File: C/hello-world.c
/**
 * @file hello-world.c
 * @brief Classic "Hello World" program in C
 * 
 * This is the traditional first program written when learning C programming.
 * It demonstrates the basic structure of a C program and the use of the
 * standard I/O library for printing to the console.
 * 
 * @author Code-Contribution Community
 * @version 1.0
 * @date 2025-10-15
 * 
 * @copyright MIT License
 * 
 * @details
 * The program consists of:
 * - Including the standard I/O header (stdio.h)
 * - Defining the main function (program entry point)
 * - Using printf() to output text
 * - Returning 0 to indicate successful execution
 */

#include <stdio.h>

/**
 * @brief Main entry point of the program
 * 
 * The main function is where program execution begins.
 * In C, the main function can be declared as:
 * - int main(void) - takes no arguments
 * - int main(int argc, char *argv[]) - accepts command-line arguments
 * 
 * @param void No parameters
 * @return int Exit status (0 indicates success)
 * 
 * @note The return value is passed to the operating system.
 *       By convention, 0 means success, non-zero indicates an error.
 * 
 * @code
 * // Compile and run:
 * gcc hello-world.c -o hello-world
 * ./hello-world
 * @endcode
 * 
 * @par Expected Output:
 * @code
 * Hello World!
 * @endcode
 */
int main(void) {
    // Print greeting to standard output
    printf("Hello World!");
    
    // Return success status to OS
    return 0;
}

File: Java/HelloWorld.java
/**
 * Hello World Program in Java
 * 
 * This is a simple Java program that demonstrates the basic syntax of Java
 * by printing "Hello World!" to the console. This is traditionally the first
 * program written when learning a new programming language.
 * 
 * <p>The program contains a single class with a main method, which serves
 * as the entry point for Java applications.</p>
 * 
 * <h2>Compilation and Execution:</h2>
 * <pre>
 * {@code
 * javac HelloWorld.java
 * java HelloWorld
 * }
 * </pre>
 * 
 * <h2>Expected Output:</h2>
 * <pre>
 * Hello World!
 * </pre>
 * 
 * @author Code-Contribution Community
 * @version 1.0
 * @since 2025-10-19
 */
public class HelloWorld {
    
    /**
     * Main method - the entry point of the Java application.
     * 
     * <p>This method is called by the Java Virtual Machine (JVM) when the
     * program is executed. It prints "Hello World!" to the standard output
     * stream (console).</p>
     * 
     * @param args Command line arguments passed to the program (not used in this example)
     * 
     * @see System#out
     * @see java.io.PrintStream#println(String)
     */
    public static void main(String[] args) {
        // Print the classic "Hello World!" message to the console
        System.out.println("Hello World!");
    }
}